charms/tensorboard-controller/src/charm.py

```diff
@@ -1,184 +1,259 @@
 #!/usr/bin/env python3
-# Copyright 2021 Canonical Ltd.
+# Copyright 2023 Canonical Ltd.
 # See LICENSE file for licensing details.
 
 import logging
-from pathlib import Path
+from typing import Dict, Tuple
 
-import yaml
+from charmed_kubeflow_chisme.exceptions import ErrorWithStatus, GenericCharmRuntimeError
+from charmed_kubeflow_chisme.kubernetes import (
+    KubernetesResourceHandler,
+    create_charm_default_labels,
+)
+from charmed_kubeflow_chisme.pebble import update_layer
 from charms.istio_pilot.v0.istio_gateway_info import GatewayRelationError, GatewayRequirer
-from oci_image import OCIImageResource, OCIImageResourceError
+from lightkube import ApiError
+from lightkube.generic_resource import load_in_cluster_generic_resources
+from lightkube.resources.apiextensions_v1 import CustomResourceDefinition
+from lightkube.resources.rbac_authorization_v1 import ClusterRole, ClusterRoleBinding
 from ops.charm import CharmBase
 from ops.main import main
-from ops.model import ActiveStatus, MaintenanceStatus, WaitingStatus
+from ops.model import ActiveStatus, Container, MaintenanceStatus, WaitingStatus
+from ops.pebble import Layer
 
+PROBE_PORT = "8081"
+PROBE_PATH = "/healthz"
+PROBE_NAME = "tensorboard-controller-up"
```

Comment:
We removed the health checks (`container.get_check()`), maybe we don't need this anymore.

Reply:
Reading a bit more into the code, you are setting `on-check-failure` in the pebble layer, so we are in fact using it. We can resolve this conversation.